### PR TITLE
Fixes #30249 - allow collection roles to be imported

### PR DESCRIPTION
**Pull request by cbergmann**
This Patch allows to import roles that are part of collections. It also imports the variables of these roles.

### CODE REVIEW

File: lib/smart_proxy_ansible/roles_reader.rb [hunk 1]
```diff
@@ -42,11 +54,25 @@ def read_roles(roles_path)
...
+        def read_collection_roles(collections_path)
+          rescue_and_raise_file_exception ReadRolesException,
+                                          collections_path, 'roles' do
+            Dir.glob("#{collections_path}/ansible_collections/*/*/roles/*").map do |path|
+              parts = path.split('/')
+              role = parts.pop
+              parts.pop
+              collection = parts.pop
+              author = parts.pop
+              author+"."+collection+"."+role
```
Comment (xprazak2) [inline lib/smart_proxy_ansible/roles_reader.rb:66]: You can use string interpolation for better style: 
```ruby
"#{author}.#{collection}.#{role}"
```

File: lib/smart_proxy_ansible/roles_reader.rb [hunk 2]
```diff
@@ -7,20 +7,32 @@ class RolesReader
       class << self
         DEFAULT_CONFIG_FILE = '/etc/ansible/ansible.cfg'.freeze
         DEFAULT_ROLES_PATH = '/etc/ansible/roles:/usr/share/ansible/roles'.freeze
+        DEFAULT_COLLECTIONS_PATHS = '/etc/ansible/collections'.freeze
```
Comment (xprazak2) [inline lib/smart_proxy_ansible/roles_reader.rb:10]: The [default paths](https://docs.ansible.com/ansible/latest/reference_appendices/config.html#collections-paths) for collections are ~/.ansible/collections:/usr/share/ansible/collections, I would add at least the second one here as well.

**Comment by xprazak2** [comment]
Works nicely, I have added a couple of comments inline. I have created a ticket in [redmine](https://projects.theforeman.org/issues/30249), could you please modify your commit message to "Fixes #30249 - $message", that should make the check for redmine issues pass.

**Comment by xprazak2** [comment]
Nice. I think the test failures might be related.

**Comment by adamruzicka** [comment]
What is the status here?

**Comment by xprazak2** [comment]
Merged as part of https://github.com/theforeman/smart_proxy_ansible/pull/31, therefore closing. Thanks!

**Pull request closed without merging**